compass/validate.py

```diff
@@ -58,38 +58,68 @@ def compare_variables(test_case, variables, filename1, filename2=None,
         Whether to check to make sure files are valid outputs of steps in
         the test case.  This should be set to ``False`` if comparing with an
         output of a step in another test case.
+
+    skip_if_step_not_run : bool, optional
+        Whether to skip the variable comparison if a user did not run one (or
+        both) of the steps involved in the comparison.  This would happen if
+        users are running steps individually or has edited ``steps_to_run``
+        in the config file to exclude one of the steps.
     """
     work_dir = test_case.work_dir
 
-    if test_case.validation is not None:
-        validation = test_case.validation
-    else:
-        validation = {'internal_pass': None,
-                      'baseline_pass': None}
-
     path1 = os.path.abspath(os.path.join(work_dir, filename1))
     if filename2 is not None:
         path2 = os.path.abspath(os.path.join(work_dir, filename2))
     else:
         path2 = None
-    if check_outputs:
-        file1_found = False
-        file2_found = False
-        for step_name, step in test_case.steps.items():
-            for output in step.outputs:
-                # outputs are already absolute paths combined with the step dir
-                if output == path1:
-                    file1_found = True
-                if output == path2:
-                    file2_found = True
 
+    file1_found = False
+    file2_found = False
+    step_name1 = None
+    step_name2 = None
+    for step_name, step in test_case.steps.items():
+        for output in step.outputs:
+            # outputs are already absolute paths combined with the step dir
+            if output == path1:
+                file1_found = True
```

Comment:
That's a good question.  No, the list of outputs are those identified by calling `add_output_file`.  There is no relationship to whether the file exists here, just whether it was identified as an output in a step.  `file1_found` doesn't literally mean that the file was found on disk but rather than the file was found as a listed output of a step in the test case, thus we know what step the file should have been created by.

Reply:
Ok, I thought that might be the case.  Thanks for clarifying.